### PR TITLE
add tests for 2905

### PR DESCRIPTION
**Pull request by andriidovhan**
covers: #3002

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy [hunk 1]
```diff
@@ -195,4 +197,94 @@ class SwitchesSpec extends HealthCheckSpecification {
...
+        when: "Install switch rules on non-existing switch"
+        northbound.installSwitchRules(NON_EXISTENT_SWITCH_ID, InstallRulesAction.INSTALL_DEFAULTS)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found"
+    }
+
+    @Ignore("is not working")
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: link to defect?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy [hunk 2]
```diff
@@ -195,4 +197,94 @@ class SwitchesSpec extends HealthCheckSpecification {
...
+
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when configuring port on non-existing switch"() {
+        when: "Configure port on non-existing switch"
+        northbound.portUp(NON_EXISTENT_SWITCH_ID, 33)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: I'm fine with how it is now, but could have been done as parameterized test
```
where:
data << [
  [descr: "delete", operation: { northbound.deleteSwitch(...) }],
  [descr: "port up", operation: { northbound.portUp(...) }]
]
```

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: I am awaiting fix from dev team

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: Only @Shared and static fields may be accessed from here.
I will leave it as is

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy [hunk 3]
```diff
@@ -195,4 +197,94 @@ class SwitchesSpec extends HealthCheckSpecification {
...
+    @Ignore("is not working")
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when deleting switch rules on non-existing switch"() {
+        when: "Delete switch rules on non-existing switch"
+        northbound.deleteSwitchRules(NON_EXISTENT_SWITCH_ID, DeleteRulesAction.DROP_ALL_ADD_DEFAULTS)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found."
```
Comment (rozdy) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: No dot will be here

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy (hunk 2 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: just do `getNorthbound()`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy (hunk 3 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy]: thx,fixed

**Pull request merged**